parser/v2/elementparser.go

```diff
@@ -422,5 +439,18 @@ func (elementParser) Parse(pi *parse.Input) (r Element, ok bool, err error) {
 		err = parse.Error(fmt.Sprintf("<%s>: %s", r.Name, strings.Join(msgs, ", ")), start)
 	}
 
+	r.LineBreak = true
+
+	for _, t := range ignoreNewLinesTags {
+		if r.Name == t {
+			r.LineBreak = false
+			break
+		}
+	}
+
+	if line, _ := pi.Peek(1); line == "\n" {
+		r.LineBreak = true
+	}
```

Comment:
This might be easier to understand if it was a function.

I think the function might be called "shouldElementHaveATrailingLineBreak"?

```suggestion
	r.LineBreak = true
	for _, t := range ignoreNewLinesTags {
		if r.Name == t {
			r.LineBreak = false
			break
		}
	}
	if line, _ := pi.Peek(1); line == "\n" {
		r.LineBreak = true
	}
```